setup.py

```diff
@@ -8,7 +7,32 @@
 from setuptools import setup
 from setuptools.command.test import test as TestCommand
 
-here = os.path.abspath(os.path.dirname(__file__))
+
+CURRENT_PYTHON = sys.version_info[:2]
+REQUIRED_PYTHON = (3, 7)
+
+if CURRENT_PYTHON < REQUIRED_PYTHON:
+    sys.stderr.write(
+        """
+==========================
+Unsupported Python version
+==========================
+This version of Requests requires Python {}.{}, but you're trying to
+install it on Python {}.{}.
+This may be because you are using a version of pip that doesn't
+understand the python_requires classifier. Make sure you
```

Comment:
We discussed this in the urllib3 Discord with @nateprewitt and @graingert and figured out that this series of steps isn't always going to solve the problem because pip *also* dropped Python versions over time.

There were a few solutions proposed:
- Keep the current message
- Always recommend installing pip<21 (last version that supported Python 2 and 3.5)
- Recommend installing [pip-with-requires-python](https://pypi.org/project/pip-with-requires-python)

I also raised the point of recommending users upgrade pip/setuptools would likely break OS installations.

Reply:
I think `pip-with-requires-python` is likely the safest option. Alternatively, we'd vend the requires values from the package in our setup.py and compose the install message from those values as @graingert suggested. That avoids any risk for Python 3.4 users who still comprise a significant number of downloads.

Reply:
@pradyunsg Noted that new Python versions come with a "good" pip version out of the box. Maybe our strategy should be to instruct the user to either pin to <2.28 or upgrade to at least Python 3.7 (since they'll need that version to install Requests anyways) and then we don't need to mention pip/setuptools at all.

Reply:
I went ahead and [ripped out the whole of touching pip versions](https://github.com/psf/requests/pull/6091/commits/febcd5199c278d99232124e22b3ab2d912b7cc16) after further discussion. If users want to upgrade their pip, that's something they can manage, otherwise the simplest option is to upgrade their Python version which will provide this automatically.

Other users are free to pin to older supported versions as discussed in the original deprecation announcement.